substrate/frame/support/src/traits/tokens/fungibles/regular.rs

```diff
@@ -363,6 +369,10 @@ pub trait Mutate<AccountId>: Inspect<AccountId> + Unbalanced<AccountId> {
 		let _extra = Self::can_withdraw(asset.clone(), source, amount)
 			.into_result(preservation != Expendable)?;
 		Self::can_deposit(asset.clone(), dest, amount, Extant).into_result()?;
+		if source == dest {
```

Comment:
Probably not important, but since the trait bound is `PartialEq` and not `Eq`, we are permitting a weird edge case where `source == dest`, but `source != source` and/or `dest != dest`. In practice, I don't think an account ID would ever _not_ be equal to itself, but just making a note here to ensure that we believe `PartialEq` is enough.

Reply:
Yea, might be saner to use `Eq` then.

Reply:
Ah I've just seen this, I think a few are missing, but that also explains why it's only in a small number of places.